### PR TITLE
Versioned Benchmarks

### PR DESCRIPTION
**Pull request by AdityaSripal**
- Creates a more detailed version struct
- Prefaces all benchmark results with version information.

Should be merged before pruning PR: #158 

Will be useful to monitor performance regressions/improvements clearly, keep older benchmark results even in new version codebase for comparisons.

### CODE REVIEW

**Comment by AdityaSripal** [comment]
TODO:

- [ ] add benchmark results from other machines
- [ ] organize benchmark results in subfolders labelled by version-commit

**Comment by marbar3778** [comment]
can #109 be closed in favor of this one?

**Comment by AdityaSripal** [comment]
#109 adds new benchmark test, while this one is to just add versioning info so they're a bit orthogonal. Though I think any future benchmark results that get merged should have version info included

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## IAVL+ Tree
 
-**Note: Requires Go 1.8+**
+**Note: Requires Go 1.11+**
```
Comment (marbar3778) [inline README.md]: 👍

**Pull request merged**